mne/tests/test_annotations.py

```diff
@@ -468,4 +473,203 @@ def test_annotations_crop():
         a.copy().crop(tmin=0, tmax=12, emit_warning=True)
 
 
+@testing.requires_testing_data
+def test_events_from_annot_in_raw_objects():
+    """Test basic functionality of events_fron_annot for raw objects."""
+    raw = read_raw_fif(fif_fname)
+    events = mne.find_events(raw)
+    event_id = {
+        'Auditory/Left': 1,
+        'Auditory/Right': 2,
+        'Visual/Left': 3,
+        'Visual/Right': 4,
+        'Visual/Smiley': 32,
+        'Motor/Button': 5
+    }
+    event_map = {v: k for k, v in event_id.items()}
+    annot = Annotations(onset=raw.times[events[:, 0] - raw.first_samp],
+                        duration=np.zeros(len(events)),
+                        description=[event_map[vv] for vv in events[:, 2]],
+                        orig_time=None)
+    raw.set_annotations(annot)
+
+    events2, event_id2 = \
+        events_from_annotations(raw, event_id=event_id, regexp=None)
+    assert_array_equal(events, events2)
+    assert_equal(event_id, event_id2)
+
+    events3, event_id3 = \
+        events_from_annotations(raw, event_id=None, regexp=None)
+
+    assert_array_equal(events[:, 0], events3[:, 0])
+    assert set(event_id.keys()) == set(event_id3.keys())
+
+    first = np.unique(events3[:, 2])
+    second = np.arange(1, len(event_id) + 1, 1).astype(first.dtype)
+    assert_array_equal(first, second)
+
+    first = np.unique(list(event_id3.values()))
+    second = np.arange(1, len(event_id) + 1, 1).astype(first.dtype)
+    assert_array_equal(first, second)
+
+    events4, event_id4 =\
+        events_from_annotations(raw, event_id=None, regexp='.*Left')
+
+    expected_event_id4 = {k: v for k, v in event_id.items() if 'Left' in k}
+    assert_equal(event_id4.keys(), expected_event_id4.keys())
+
+    expected_events4 = events[(events[:, 2] == 1) | (events[:, 2] == 3)]
+    assert_array_equal(expected_events4[:, 0], events4[:, 0])
+
+    events5, event_id5 = \
+        events_from_annotations(raw, event_id=event_id, regexp='.*Left')
+
+    expected_event_id5 = {k: v for k, v in event_id.items() if 'Left' in k}
+    assert_equal(event_id5, expected_event_id5)
+
+    expected_events5 = events[(events[:, 2] == 1) | (events[:, 2] == 3)]
+    assert_array_equal(expected_events5, events5)
+
+    with pytest.raises(ValueError, match='not find any of the events'):
+        events_from_annotations(raw, regexp='not_there')
+
+    raw.set_annotations(None)
+    events7, _ = events_from_annotations(raw)
+    assert_array_equal(events7, np.empty((0, 3), dtype=int))
+
+
+def test_events_from_annot_onset_alingment():
+    """Test events and annotations onset are the same."""
+    from mne.io.tests.test_raw import _raw_annot
+    #       sec  0        1        2        3
+    #       raw  .        |--------XXXXXXXXX
+    #     annot  .             |---XX
+    # raw.annot  .        |--------XX
+    #   latency  .        0        1        2
+    #            .                 0        0
```

Comment:
So based on https://github.com/mne-tools/mne-python/pull/5525#discussion_r219150780 this import should not be here at all and some CI should have complained.

Reply:
@larsoner

Reply:
It does not test if things are too nested (like this is), it just checks that things are nested enough (such as slow `scipy` submodule imports being nested). Also the `test_*` file imports do not need to be nested, nothing is imported from `test_*` files when you `import mne`.